diff --git a/src/util.c b/src/util.c
index 10485db..8f702dd 100644
--- a/src/util.c
+++ b/src/util.c
@@ -14,11 +14,13 @@ unsigned int gcd(unsigned int a, unsigned int b) {
 
 size_t truncate_utf8_string(char* const buffer, const size_t size,
                             const size_t char_limit) {
+    unsigned short ellipsis = 1;
     size_t char_count = 0;
     size_t i = 0;
     while (char_count < char_limit) {
         char ch = buffer[i];
         if (ch == '\0') {
+            ellipsis = 0;
             break;
         }
 
@@ -43,6 +45,9 @@ size_t truncate_utf8_string(char* const buffer, const size_t size,
         i += skip;
     }
 
+    if (ellipsis) {
+      buffer[i-3] = buffer[i-2] = buffer[i-1] = '.';
+    }
     buffer[i] = '\0';
 
     return i + 1;
