diff --git a/config.h b/config.h
index 0dbb2b0..b8a3b6b 100644
--- a/config.h
+++ b/config.h
@@ -2,7 +2,7 @@
 #define CONFIG_H
 
 // String used to delimit block outputs in the status.
-#define DELIMITER "  "
+#define DELIMITER " • "
 
 // Maximum number of Unicode characters that a block can output.
 #define MAX_BLOCK_OUTPUT_LENGTH 45
@@ -17,17 +17,14 @@
 #define TRAILING_DELIMITER 0
 
 // Define blocks for the status feed as X(icon, cmd, interval, signal).
-#define BLOCKS(X)             \
-    X("", "sb-mail", 600, 1)  \
-    X("", "sb-music", 0, 2)   \
-    X("", "sb-disk", 1800, 3) \
-    X("", "sb-memory", 10, 4) \
-    X("", "sb-swap", 10, 0)   \
-    X("", "sb-loadavg", 5, 5) \
-    X("", "sb-mic", 0, 6)     \
-    X("", "sb-record", 0, 7)  \
-    X("", "sb-volume", 0, 8)  \
-    X("", "sb-battery", 5, 9) \
-    X("", "sb-date", 1, 10)
+#define BLOCKS(X)                  \
+    X("", "sb-now-playing", 1, 7)  \
+    X("", "sb-nettraf", 1, 6)      \
+    X("", "sb-brightness", 0, 5)   \
+    X("", "sb-battery", 30, 8)     \
+    X("", "sb-audio-sink", 0, 4)   \
+    X("", "sb-audio-source", 0, 3) \
+    X("", "sb-memory", 1, 2)       \
+    X("", "sb-clock", 60, 1)
 
 #endif  // CONFIG_H
